src/checks/json-schema-errors.ts

```diff
@@ -33,8 +33,18 @@ export default function jsonSchemaErrors(definition: StateMachine): StateMachine
         ],
         allowUnionTypes: true
     });
-    registerAll(ajv);
-    ajv.addFormat("asl_arn", isArnFormatValid)
+    if (options.checkPaths) {
+        registerAll(ajv);
+    } else {
+        ajv.addFormat("asl_path", () => true);
+        ajv.addFormat("asl_ref_path", () => true);
+        ajv.addFormat("asl_payload_template", () => true);
+    }
+    if (options.checkArn) {
+        ajv.addFormat("asl_arn", isArnFormatValid)
+    } else {
+        ajv.addFormat("asl_arn", () => true);
+    }
```

Comment:
these are the flags in action for now

if there's more elaborate control needed over the validation then it may require modifying the input shapes for AJV. In general, I think it's a better approach to tweak the schema definition and allow the AJV engine to implement the validation for us.